lib/trino-plugin-toolkit/src/test/java/io/trino/plugin/base/security/TestFileBasedSystemAccessControl.java

```diff
@@ -1370,7 +1384,59 @@ public void testRefreshing()
     public void parseUnknownRules()
     {
         assertThatThrownBy(() -> newFileBasedSystemAccessControl("file-based-system-security-config-file-with-unknown-rules.json"))
-                .hasMessageContaining("Invalid JSON");
+                .hasMessageContaining("Failed to parse JSON");
+    }
+
+    @Test
+    public void noHttpClientConfigForLocalFile()
+    {
+        FileBasedSystemAccessControl.Factory factory = new FileBasedSystemAccessControl.Factory();
+        SystemAccessControl accessControl = factory.create(ImmutableMap.of(
+                "security.config-file", getResourcePath("file-based-system-catalog.json")));
+        SystemSecurityContext alice = new SystemSecurityContext(TestFileBasedSystemAccessControl.alice, queryId);
+        accessControl.checkCanCreateView(alice, aliceView);
+        AccessControlRulesProvider<FileBasedSystemAccessControlRules> provider = factory.injector.getInstance(
+                Key.get(new TypeLiteral<AccessControlRulesProvider<FileBasedSystemAccessControlRules>>() {}));
+        assertTrue(provider instanceof LocalFileBasedSystemAccessControlRulesProvider);
+        assertThatThrownBy(() -> factory.injector.getInstance(Key.get(HttpClient.class, ForAccessControlRules.class)))
+                .isInstanceOf(ConfigurationException.class).hasMessageContaining("is not explicitly bound");
+        assertThatThrownBy(() -> factory.injector.getInstance(Key.get(HttpClientConfig.class)))
+                .isInstanceOf(ConfigurationException.class).hasMessageContaining("is not explicitly bound");
+    }
+
+    @Test
+    public void httpClientConfigForRestFile()
+            throws IOException
+    {
+        HttpServer httpServer = HttpServer.create(new InetSocketAddress(2002), 0);
```

Comment:
Please extract a `TestingHttpServer` that is closeable. Then use plain try-with-resources.